File: 0674-longest-continuous-increasing-subsequence/0674-longest-continuous-increasing-subsequence.cpp
class Solution {
public:
    int findLengthOfLCIS(vector<int>& nums) {
      int current_r = 0, best_r = 0;
      for (int i = 0; i < nums.size(); i++){
        if (i == 0 || (nums[i-1] >= nums[i])){
          /* reset current_r */
          current_r = 1;
        } else{
            /* increase 1 in current_r */
          current_r++;
        }
        best_r = max(best_r, current_r);
      }
      return best_r;
    }
};

File: 0674-longest-continuous-increasing-subsequence/0674-longest-continuous-increasing-subsequence.py
class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        current_r: int = 0
        best_r: int = 0
        for i in range(len(nums)):
            if i == 0 or nums[i-1] >= nums[i]:
                current_r = 1
            else:
                current_r += 1
            best_r = max(best_r, current_r)
        return best_r
